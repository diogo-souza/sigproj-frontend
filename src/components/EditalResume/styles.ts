import { darken } from 'polished';
import styled, { css } from 'styled-components';

export const Header = styled.div`
  ${({ theme }) => css`
    font-weight: ${theme.font.bold};
    border-bottom: 1px solid ${theme.colors.gray};
    > h3 {
      font-size: ${theme.font.sizes.xxlarge};
    }
    > p {
      color: ${darken(0.2, theme.colors.gray)};
    }

    @media (max-width: 768px) {
      > h3 {
        font-size: ${theme.font.sizes.xlarge};
      }
      > p {
        font-size: ${theme.font.sizes.medium};
      }
    }
  `}
`;

export const Content = styled.dl`
  ${({ theme }) => css`
    > div {
      display: grid;
      grid-template-columns: 1fr 2fr;
      & + div {
        border-top: 1px solid ${theme.colors.gray};
      }

      @media (max-width: 768px) {
        display: block;
      }
    }
  `}
`;

export const Field = styled.dt`
  ${({ theme }) => css`
    color: ${darken(0.2, theme.colors.gray)};
  `}
`;

export const Modalities = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-wrap: wrap;
    flex: 1 0 50%;
    color: ${theme.colors.titleWhite};
    p {
      font-size: ${theme.font.sizes.small};
      margin-right: ${theme.spacings.xxsmall};
      background-color: ${darken(0.075, theme.colors.secondary)};
      border-radius: ${theme.border.radius};
      padding: calc(5.6px * 0.9) calc(10.4px * 0.9);
      margin-top: 8px;

      &:last-child {
        margin-right: 0;
      }
    }
  `}
`;

export const Data = styled.dd`
  ${({ theme }) => css`
    color: ${darken(0.2, theme.colors.titleBlack)};
    font-weight: ${theme.font.bold};

    > div {
      font-weight: ${theme.font.semiBold};
      font-weight: ;
      h2 {
        font-weight: ${theme.font.bold};
        font-size: calc(${theme.font.sizes.large} * 2);
        margin-bottom: 2.5rem;
      }
      p {
        text-align: justify;
      }

      > div + div {
        margin-top: ${theme.spacings.xsmall};
      }
    }

    @media (max-width: 768px) {
      margin-top: calc(${theme.spacings.xxsmall} / 2);
    }

    ${Modalities} {
      margin-top: -8px;
    }
  `}
`;

export const Container = styled.div`
  ${({ theme }) => css`
    border: 1px solid ${theme.colors.gray};
    border-radius: ${theme.border.radius};
    background-color: ${theme.colors.white};
    ${Header} {
      padding: calc(${theme.spacings.small} - 4px) ${theme.spacings.small};
    }
    ${Content} {
      > * {
        padding: calc(${theme.spacings.small} - 4px) ${theme.spacings.small};
      }
    }
  `}
`;
