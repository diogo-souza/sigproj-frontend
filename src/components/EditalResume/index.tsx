/* eslint-disable react/no-danger */
import React from 'react';

import { EditalData } from 'types/editalTypes';

import * as S from './styles';

type EditalResumeProps = {
  edital: EditalData;
};

const EditalResume: React.FC<EditalResumeProps> = ({
  edital,
}: EditalResumeProps) => {
  return (
    <S.Container>
      <S.Header>
        <h3>Informações do Edital</h3>
        <p>Edital de Apoio ou de Fluxo Contínuo</p>
      </S.Header>
      <S.Content>
        <div>
          <S.Field>Título</S.Field>
          <S.Data>{edital.titulo}</S.Data>
        </div>
        <div>
          <S.Field>Chamada</S.Field>
          <S.Data>{edital.chamada}</S.Data>
        </div>
        <div>
          <S.Field>Data de inicio</S.Field>
          <S.Data>{edital.data_inicio}</S.Data>
        </div>
        <div>
          <S.Field>Data final</S.Field>
          <S.Data>{edital.data_fim}</S.Data>
        </div>
        <div>
          <S.Field>Modalidades</S.Field>
          <S.Data>
            {edital.modalidades && (
              <S.Modalities>
                {edital.modalidades.map(modalidade => (
                  <p key={modalidade}>{modalidade}</p>
                ))}
              </S.Modalities>
            )}
          </S.Data>
        </div>
        <div>
          <S.Field>Corpo</S.Field>
          <S.Data>
            <div
              dangerouslySetInnerHTML={{
                __html:
                  edital?.corpo ||
                  `<p><strong>Não existe corpo para o edital</strong><p>`,
              }}
            />
          </S.Data>
        </div>
      </S.Content>
    </S.Container>
  );
};

export default EditalResume;
