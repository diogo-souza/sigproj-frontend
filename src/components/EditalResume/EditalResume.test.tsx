import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import EditalResume from '.';

describe('<EditalResume />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(
      <EditalResume
        edital={{
          uuid: 'id-1',
          titulo: 'Title 1',
          ativo: true,
          data_inicio: '19/08/2021',
          data_fim: '20/08/2021',
          chamada: 'headline 1',
          modalidades: ['Programa', 'Projeto'],
          corpo: '<p>Insira o corpo do edital aqui</p>\n',
        }}
      />,
    );
    expect(screen.getByText(/title 1/i)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
