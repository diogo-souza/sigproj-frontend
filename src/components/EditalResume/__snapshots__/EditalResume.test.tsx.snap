// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditalResume /> should render the heading 1`] = `
.c2 {
  font-weight: 600;
  border-bottom: 1px solid #939393;
}

.c2 > h3 {
  font-size: 2.8rem;
}

.c2 > p {
  color: #606060;
  font-size: 1.6rem;
}

.c4 > div {
  display: grid;
  grid-template-columns: 1fr 2fr;
}

.c4 > div + div {
  border-top: 1px solid #939393;
}

.c5 {
  color: #606060;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 1 0 50%;
  -ms-flex: 1 0 50%;
  flex: 1 0 50%;
  color: #FFFFFF;
}

.c8 p {
  font-size: 1.4rem;
  margin-right: 0.8rem;
  background-color: #117d8c;
  border-radius: 0.4rem;
  padding: calc(5.6px * 0.9) calc(10.4px * 0.9);
  margin-top: 8px;
}

.c8 p:last-child {
  margin-right: 0;
}

.c6 {
  color: #000;
  font-weight: 600;
}

.c6 > div {
  font-weight: 500;
}

.c6 > div h2 {
  font-weight: 600;
  font-size: calc(1.8rem * 2);
  margin-bottom: 2.5rem;
}

.c6 > div p {
  text-align: justify;
}

.c6 > div > div + div {
  margin-top: 1.6rem;
}

.c6 .c7 {
  margin-top: -8px;
}

.c0 {
  border: 1px solid #939393;
  border-radius: 0.4rem;
  background-color: #FFFFFF;
}

.c0 .c1 {
  padding: calc(2.4rem - 4px) 2.4rem;
}

.c0 .c3 > * {
  padding: calc(2.4rem - 4px) 2.4rem;
}

@media (max-width:768px) {
  .c2 > h3 {
    font-size: 2.0rem;
  }
}

@media (max-width:768px) {
  .c4 > div {
    display: block;
  }
}

@media (max-width:768px) {
  .c6 {
    margin-top: calc(0.8rem / 2);
  }
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  >
    <h3>
      Informações do Edital
    </h3>
    <p>
      Edital de Apoio ou de Fluxo Contínuo
    </p>
  </div>
  <dl
    class="c3 c4"
  >
    <div>
      <dt
        class="c5"
      >
        Título
      </dt>
      <dd
        class="c6"
      >
        Title 1
      </dd>
    </div>
    <div>
      <dt
        class="c5"
      >
        Chamada
      </dt>
      <dd
        class="c6"
      >
        headline 1
      </dd>
    </div>
    <div>
      <dt
        class="c5"
      >
        Data de inicio
      </dt>
      <dd
        class="c6"
      >
        19/08/2021
      </dd>
    </div>
    <div>
      <dt
        class="c5"
      >
        Data final
      </dt>
      <dd
        class="c6"
      >
        20/08/2021
      </dd>
    </div>
    <div>
      <dt
        class="c5"
      >
        Modalidades
      </dt>
      <dd
        class="c6"
      >
        <div
          class="c7 c8"
        >
          <p>
            Programa
          </p>
          <p>
            Projeto
          </p>
        </div>
      </dd>
    </div>
    <div>
      <dt
        class="c5"
      >
        Corpo
      </dt>
      <dd
        class="c6"
      >
        <div>
          <p>
            Insira o corpo do edital aqui
          </p>
          

        </div>
      </dd>
    </div>
  </dl>
</div>
`;
